packages/timelock-manager/contracts/TimeLockManagerReversible.sol

```diff
@@ -0,0 +1,260 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@api3-contracts/api3-token/contracts/interfaces/IApi3Token.sol";
+import "./interfaces/ITimelockManagerReversible.sol";
+
+/// @title Contract that the API3 DAO uses to timelock API3 tokens
+/// @notice The owner of TimelockManager (i.e., API3 DAO) can send tokens to
+/// TimelockManager to timelock them. These tokens will then be vested to their
+/// recipient linearly, starting from releaseStart and ending at releaseEnd of
+/// the respective timelock.
+
+contract TimelockManagerReversible is Ownable, ITimelockManagerReversible {
+    using SafeMath for uint256;
+
+    struct Timelock {
+        uint256 totalAmount;
+        uint256 remainingAmount;
+        uint256 releaseStart;
+        uint256 releaseEnd;
+    }
+
+    IApi3Token public immutable api3Token;
+    mapping(address => Timelock) public timelocks;
+
+    /// @param api3TokenAddress Address of the API3 token contract
+    /// @param timelockManagerOwner Address that will receive the ownership of
+    /// the TimelockManager contract (i.e., the API3 DAO)
+    constructor(address api3TokenAddress, address timelockManagerOwner) public {
+        api3Token = IApi3Token(api3TokenAddress);
+        transferOwnership(timelockManagerOwner);
+    }
+
+    /// @notice Called by the owner (i.e., API3 DAO) to revert the timelock of
+    /// a recipient
+    /// @param recipient Original recipient of tokens
+    /// @param destination Destination of the tokens locked by the reverted
+    /// timelock
+    function revertTimelock(address recipient, address destination)
+        external
+        override
+        onlyOwner
+        onlyIfRecipientHasRemainingTokens(recipient)
+    {
+        require(destination != address(0), "Invalid destination");
+        uint256 remaining = timelocks[recipient].remainingAmount;
+        timelocks[recipient].remainingAmount = 0;
+        require(
+            api3Token.transfer(destination, remaining),
+            "API3 token transfer failed"
+        );
+        emit RevertedTimelock(recipient, destination, remaining);
+    }
+
+    /// @notice Called by the owner (i.e., API3 DAO) to stop the vesting
+    /// a recipient
+    /// @param recipient Original recipient of tokens
+    /// @param destination Destination of the excess tokens vested to the addresss
+    function stopVesting(address recipient, address destination)
```

Comment:
my solidity formatter in vs code, ended up doing this, will disable it going forward.

Reply:
Which one is that? We can pick one and agree on a common configuration.

Reply:
I am actually using the solidity plugin in vscode that comes with solhint linter and prettier plugin for code formatting. [Link to the plugin](https://github.com/juanfranblanco/vscode-solidity#formatting-using-prettier-and-the-prettier-solidity-plugin). I am not sure if it would be possible to enforce one line per argument in the formatter, that's why I disabled it and manually reformatted it.

Reply:
Created an issue
https://github.com/api3dao/airnode/issues/247